clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsCramWebServiceSpec.scala

```diff
@@ -0,0 +1,200 @@
+      status shouldEqual StatusCodes.OK
+      memorySearchDAO.updateCalls should have length 1
+      val firstUpdate = memorySearchDAO.updateCalls.headOption
+        .map(_._1)
+        .getOrElse {
+          // Doing this .headOption.getOrElse dance because Codacy
+          // scolds us for using .head
+          fail("Impossible because of the above check")
+        }
+        .asInstanceOf[DocumentWgsCram]
+
+      firstUpdate.upsertId should be(responseAs[UpsertId])
+      firstUpdate.cramMd5 should be(Some("abcgithashdef"))
+      firstUpdate.notes should be(Some("some note"))
+      firstUpdate.cramPath should be(Some("gs://path/cram.cram"))
+    }
+
+    // We have to test the MemorySearchDAO because we're not going to implement
+    // Elasticsearch logic in our test specs. Here, we're just verifying that
+    // the web service passes the appropriate queries onto the search DAO.
+    Post("/query", Map("location" -> "GCP")) ~> webService.cramQuery ~> check {
+      memorySearchDAO.queryCalls should be(
+        Seq(
+          TransferWgsCramV1QueryInput(
+            location = Some(Location.GCP),
+            documentStatus = Some(DocumentStatus.Normal)
+          )
+        ).map(WgsCramService.v1QueryConverter.buildQuery)
+      )
+    }
+
+    Post(
+      "/metadata/GCP/proj0/alias/2",
+      Map(
+        "cram_md5" -> "abcgithashdef",
+        "notes" -> "some note",
+        "document_status" -> "Deleted",
+        "cram_path" -> ""
+      )
+    ) ~> webService.cramPostMetadata ~> check {
+      status shouldEqual StatusCodes.OK
+      memorySearchDAO.updateCalls should have length 2
+      val secondUpdate = memorySearchDAO.updateCalls
+        .map(_._1)
+        .apply(1)
+        .asInstanceOf[DocumentWgsCram]
+      secondUpdate.cramMd5 should be(Some("abcgithashdef"))
+      secondUpdate.notes should be(Some("some note"))
+      secondUpdate.documentStatus should be(Some(DocumentStatus.Deleted))
+      secondUpdate.cramPath should be(Some(""))
+    }
+
+    // We have to test the MemorySearchDAO because we're not going to implement
+    // Elasticsearch logic in our test specs. Here, we're just verifying that
+    // the web service passes the appropriate queries onto the search DAO.
+    Post("/queryall", Map("location" -> "GCP")) ~> webService.cramQueryall ~> check {
+      memorySearchDAO.queryCalls should be(
+        Seq(
+          // From /query call earlier in the test.
+          TransferWgsCramV1QueryInput(
+            location = Some(Location.GCP),
+            documentStatus = Some(DocumentStatus.Normal)
+          ),
+          // No documentStatus restriction from /queryall
+          TransferWgsCramV1QueryInput(location = Some(Location.GCP))
+        ).map(WgsCramService.v1QueryConverter.buildQuery)
+      )
+    }
+  }
+
+  it should "query with a BoGuS project and sample and return nothing" in {
+    val webService = new MockWgsCramWebService()
+    Post(
+      "/query",
```

Comment:
Our code shouldn't duplicate magic strings such as `query` and `metadata` and `project` and `sample_alias`. The standard endpoint path names should be defined in a base class, such as `JsonWebService`. The tests should use those constants as well.

Reply:
I think the strings are OK, and not sure using JsonWebService constants in the test code is better than "duplicating" the strings, but I don't feel too strongly about it. (Will explain myself if you want.)